mars-rover-new/src/mars_rover_new/core.clj

```diff
@@ -1,7 +1,118 @@
+;; Mars Rover Exercise
+
+;; todo: input validation, check invalid moves
+
 (ns mars-rover-new.core
+  (:require [clojure.string :as string])
   (:gen-class))
 
+(def valid-directions #{"N" "E" "S" "W"})
+(def valid-cmds #{"L" "R" "M"})
+
+(def default-mars-rover-data {:plateau-size {:x nil
+                                             :y nil}
+                              :rover-position {:x nil
+                                               :y nil
+                                               :direction nil}
+                              :cmd-sequence nil})
+
+(defn get-input
+  "Prompt (if provided) and read one line of input stream. Trim and convert to upper-case."
+  ([] (get-input ""))
+  ([prompt]
+   (when (not-empty prompt)
+     (println prompt))
+   (let [input (string/trim (read-line))]
+     (string/upper-case input))))
+
+(defn split-string
+  "Split string on white-space and remove empty elements"
+  [s]
+  (filter #(not-empty %) (string/split s #" ")))
+
+(defn str->int
+  "Convert string to integer. If invalid string, return nil"
+  [s]
+  (try (Integer/parseInt s)
+       (catch Exception e nil)))
+
+(defn in-range?
+  "Check if val is within bounds (inclusive)"
+  [val lower-bound upper-bound]
+  (if (and (>= val lower-bound) (<= val upper-bound))
```

Comment:
@shahn95 `<=` can work with more than 2 args.

So `(and (>= val lower-bound) (<= val upper-bound))` can be compressed into `(<= lower-bound val upper-bound)`